Authentication-Results: mail-a.sr.ht; dkim=pass header.d=gmail.com header.i=@gmail.com
Received: from mail-pl1-f173.google.com (mail-pl1-f173.google.com [209.85.214.173])
	by mail-a.sr.ht (Postfix) with ESMTPS id D114720322
	for <~mpu/qbe@lists.sr.ht>; Mon, 13 Jan 2025 06:08:50 +0000 (UTC)
Received: by mail-pl1-f173.google.com with SMTP id d9443c01a7336-216728b1836so65957085ad.0
        for <~mpu/qbe@lists.sr.ht>; Sun, 12 Jan 2025 22:08:50 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1736748529; x=1737353329; darn=lists.sr.ht;
        h=content-transfer-encoding:mime-version:message-id:date:subject:cc
         :to:from:from:to:cc:subject:date:message-id:reply-to;
        bh=Qj8atNHA6tFV2ohxnbaH+XftNFWb7lSZQy/niOda0NE=;
        b=VDhcx5RjTERTeyO6F+enlhN1ghT3CBUlKaxDmYT2vENCaVkoYWsgyrPDJ4UNWOxRdo
         hPOM80nz9629x0jMPsCPaNA3o0lhCLPdJ6fGP7vQyGljxdHeaUbtLgkvAb57tdK8Ggpf
         pqma/DPQI8JpcWHyo3wlAtYuuSpUM7rIhys7sfkZsCiflV36pS2cNGPAU8Nl/kPMcd92
         IJirNrjPgAZOXfvbdvc46C6pPegX/Slg239bBWr9O+wgsZvDrnu/67KfvClRp+3lijyF
         e33LEt7E2iP3gXC9XTw3eOsWM9DLhZ0/Daec0zxSps1yas60QXoL3pWq0P3OnXrnCDBV
         6hUg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1736748529; x=1737353329;
        h=content-transfer-encoding:mime-version:message-id:date:subject:cc
         :to:from:x-gm-message-state:from:to:cc:subject:date:message-id
         :reply-to;
        bh=Qj8atNHA6tFV2ohxnbaH+XftNFWb7lSZQy/niOda0NE=;
        b=NqMN9TspVXe0BWBL0uVSYVS+zUBFzZbgSQXpYHCg0qz6762xaPspn5c8+QUv0REH58
         aEbqeYo8mO5tpowVSqgCsNGyjlmvLeGIMnM5eOnMw8P0Oc73B/2j+2f7ULpaBtVZ/GQX
         6HCB97ZAkR3RWn1QvtLIKQ0T5SHkN1r0bL/ep84WwXNH0J5X1At+beWsMCfHswnS//V7
         MbzNLc4HKQQp56QTDTZ+IuQXiPIMaa289hHVUVQUWD/hrgko5a8LIBUMSHoffyA+xbhX
         9gw5/0uM0D7D4XWErbBosGi3jTimldq4y/1Al7rJ3eGZpCuYWFqW9MwXQQ1cKFE34Las
         AuBg==
X-Gm-Message-State: AOJu0YzdyiC1kfTpn4k5bsw0geWc3pKG26tJ7vlXoXsJa5xvTHoJ49/T
	bREkW8V+o7ki4OMxKRC1eWF7mU6ZaX1VXRp2QK7sa9/qPZWSD7o3OfgawTPk
X-Gm-Gg: ASbGncsS7MQKhc0ZqfGcWPcsh0FHc/lgzjQPAcUoQuR9wZb2NchXMOIKV1pyQOCLB2x
	zMTMc0onM9uaunRBwi44e+IBZbZMMSdECCFMK5L/2cwpSGbztkWZw0MmXXXjtPOOIWs/m1/nBlb
	dA4Lw1oi/nBqIb879Xfw0Ibu0AbuWNkQ+dOD0AZzI436Z75a9ppVE4W6YWu5BqGGMWnhJj1jfqb
	LzvQC/hqOymGGJjt/MByhLXkXapghvfnadtpCyOjL2/w2WJPqfM3VcLWwelyL/AH+uvdNIhuK29
	6Z0WRCxDYFnCzQjIH4y1Ym9+WmCdUxM=
X-Google-Smtp-Source: AGHT+IGMVXp/mCgkuA905gT+cVYZCqVJ3C43Fo2cPCY57gd2aT1F5EOOzHErDx2DA2U89hhk1YBqjg==
X-Received: by 2002:a17:903:3385:b0:216:1cf8:8b8 with SMTP id d9443c01a7336-21a83f69643mr203344015ad.27.1736748528859;
        Sun, 12 Jan 2025 22:08:48 -0800 (PST)
Received: from localhost.localdomain (d154-5-57-197.bchsia.telus.net. [154.5.57.197])
        by smtp.gmail.com with ESMTPSA id d9443c01a7336-21a9f219f0dsm46583885ad.139.2025.01.12.22.08.47
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Sun, 12 Jan 2025 22:08:48 -0800 (PST)
From: Scott Graham <sgraham@gmail.com>
To: ~mpu/qbe@lists.sr.ht
Cc: Scott Graham <sgraham@gmail.com>
Subject: [PATCH] Fix warnings from clang about unprototyped vs (void)
Date: Sun, 12 Jan 2025 22:08:28 -0800
Message-ID: <20250113060828.784-1-sgraham@gmail.com>
X-Mailer: git-send-email 2.47.0.vfs.0.3
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

e.g.

clang -O3 -std=c99 -g -Wall -Wextra -Wpedantic -c cfg.c -o cfg.o
cfg.c:4:7: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
    4 | newblk()
      |       ^
      |        void
1 warning generated.
---
 cfg.c   |  2 +-
 parse.c | 18 +++++++++---------
 rega.c  |  2 +-
 util.c  |  2 +-
 4 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/cfg.c b/cfg.c
index 406c307..f1fa21a 100644
--- a/cfg.c
+++ b/cfg.c
@@ -1,7 +1,7 @@
 #include "all.h"
 
 Blk *
-newblk()
+newblk(void)
 {
 	static Blk z;
 	Blk *b;
diff --git a/parse.c b/parse.c
index e896679..e321254 100644
--- a/parse.c
+++ b/parse.c
@@ -179,7 +179,7 @@ err(char *s, ...)
 }
 
 static void
-lexinit()
+lexinit(void)
 {
 	static int done;
 	int i;
@@ -201,7 +201,7 @@ lexinit()
 }
 
 static int64_t
-getint()
+getint(void)
 {
 	uint64_t n;
 	int c, m;
@@ -222,7 +222,7 @@ getint()
 }
 
 static int
-lex()
+lex(void)
 {
 	static char tok[NString];
 	int c, i, esc;
@@ -332,7 +332,7 @@ Alpha:
 }
 
 static int
-peek()
+peek(void)
 {
 	if (thead == Txxx)
 		thead = lex();
@@ -340,7 +340,7 @@ peek()
 }
 
 static int
-next()
+next(void)
 {
 	int t;
 
@@ -350,7 +350,7 @@ next()
 }
 
 static int
-nextnl()
+nextnl(void)
 {
 	int t;
 
@@ -415,7 +415,7 @@ tmpref(char *v)
 }
 
 static Ref
-parseref()
+parseref(void)
 {
 	Con c;
 
@@ -580,7 +580,7 @@ findblk(char *name)
 }
 
 static void
-closeblk()
+closeblk(void)
 {
 	curb->nins = curi - insb;
 	idup(&curb->ins, insb, curb->nins);
@@ -1012,7 +1012,7 @@ parsefields(Field *fld, Typ *ty, int t)
 }
 
 static void
-parsetyp()
+parsetyp(void)
 {
 	Typ *ty;
 	int t, al;
diff --git a/rega.c b/rega.c
index 8e601c9..4d2a06b 100644
--- a/rega.c
+++ b/rega.c
@@ -247,7 +247,7 @@ pmrec(enum PMStat *status, int i, int *k)
 }
 
 static void
-pmgen()
+pmgen(void)
 {
 	int i;
 	enum PMStat *status;
diff --git a/util.c b/util.c
index 2e4b4cc..1a71030 100644
--- a/util.c
+++ b/util.c
@@ -90,7 +90,7 @@ alloc(size_t n)
 }
 
 void
-freeall()
+freeall(void)
 {
 	void **pp;
 
-- 
2.47.0.vfs.0.3

